sympy/utilities/lambdify.py

```diff
@@ -903,6 +905,59 @@ def _module_present(modname, modlist):
             return True
     return False
 
+def _cse_del(expr):
+    """
+    Same as ``cse`` but append an information on the useless vars.
+
+    expr : sympy.core.basic.Basic
+        This is the expression that we want to optimize.
+
+    Returns
+    =======
+
+    replacements : list of (Symbol, expression) pairs
+        All of the common subexpressions that were replaced.
+    useless : list of set of Symbols
+        These are the symbols that no longer appear afterwards.
+        They can therefore be deleted while running in order to relieve the RAM.
+    reduced_exprs : list of sympy expressions
+        The reduced expressions with all of the replacements above.
+
+    Examples
+    ========
+
+    >>> from sympy.utilities.lambdify import _cse_del
+    >>> from sympy import sqrt, cos, sin, Symbol
+    >>> x = Symbol("x")
+    >>> expr = sqrt((x + cos(x))*sin(x + cos(x))) + (x + cos(x))*sin(x + cos(x))
+    >>> _cse_del(expr)
+    ([(x0, x + cos(x)), (x1, x0*sin(x0))], [set(), {x0}], sqrt(x1) + x1)
+    """
+    def get_free_symbols(expr):
+        if hasattr(expr, "__iter__"):
+            return set.union(*map(get_free_symbols, expr))
+        return expr.free_symbols
+
+    from sympy import cse
+    replacements, reduced_exprs = cse([expr])
+    reduced_exprs = reduced_exprs.pop() # No raison to cast in a list!
```

Comment:
you recognize the need for allowing `expr` to be an iterable in `get_free_symbols`...you should not pop only the first -- and not necessarily the only -- expression.

Reply:
as 'cse' seemed to put the expression in a list, even though the starting expression was not an iterable, I thought that putting the item in a list from the start would just ensure consistency . But indeed, as you just pointed out to me, this cast is not systematic, so if we put the element in a list, it does not return a list of list but a simple list. It's curious but it's like that! So I corrected the problem.